reactor-core/src/test/java/reactor/core/scheduler/BoundedElasticSchedulerTest.java

```diff
@@ -1551,6 +1551,88 @@ void toStringOfExecutorReflectsIdleVsActive() throws InterruptedException {
 		}
 	}
 
+	@Test
+	public void moveToIdleAfterHaveRejectTasksAndCompleteAllTasks() throws InterruptedException {
+		AtomicInteger taskRejected = new AtomicInteger();
+
+		int maxThreads = 2;
+		int maxTaskQueuedPerThread = 10;
+
+		BoundedElasticScheduler scheduler = afterTest.autoDispose(
+				new BoundedElasticScheduler(
+						maxThreads,
+						maxTaskQueuedPerThread,
+						new ReactorThreadFactory(
+								"moveToIdleAfterHaveRejectTasksAndCompleteAllTasks",
+								new AtomicLong(),
+								false,
+								false,
+								null),
+						60));
+		scheduler.start();
+
+		CountDownLatch latch = new CountDownLatch(1);
+		CountDownLatch tasksStartedLatch = new CountDownLatch(maxThreads);
+
+		Runnable startedAndWaitLatchRunnable = () -> {
+			try {
+				tasksStartedLatch.countDown();
+				latch.await(30, TimeUnit.SECONDS);
+			}
+			catch (InterruptedException e) {
+				e.printStackTrace();
+			}
+		};
+
+		Runnable waitLatchRunnable = () -> {
+			try {
+				latch.await(30, TimeUnit.SECONDS);
+			}
+			catch (InterruptedException e) {
+				e.printStackTrace();
+			}
+		};
+
+		//initial task that blocks the threads, causing other tasks to enter pending queue
```

Comment:
//initial task that blocks the threads, causing other tasks to enter pending queue

->

    // initial tasks that block the threads, causing other tasks to enter the pending queue